swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SettingsView.swift

```diff
@@ -93,6 +99,18 @@ class SettingsViewModel: ObservableObject {
         self.updateDerivedState()
       }
       .store(in: &cancellables)
+
+    // Update our state from configuration updates
+    store.$configuration
+      .receive(on: RunLoop.main)
+      .sink { [weak self] newConfiguration in
+        self?.isAuthURLOverridden = newConfiguration?.isOverridden(Configuration.Keys.authURL) ?? false
+        self?.isApiURLOverridden = newConfiguration?.isOverridden(Configuration.Keys.apiURL) ?? false
+        self?.isLogFilterOverridden = newConfiguration?.isOverridden(Configuration.Keys.logFilter) ?? false
+
+        self?.updateDerivedState()
+      }
+      .store(in: &cancellables)
   }
 
   private func updateDerivedState() {
```

Comment:
It seems a bit odd to me that we need to persist this derived state but I am also not understanding SwiftUI good enough here. Intuitively, I would have thought that there needs to be a way to create a computed property / function that on-the-fly computes whether something is disabled _based on_ the actual state we have in the background.

Reply:
That's what the combiner does in the initializer. Unfortunately it's a bit manual if we want to support macOS 12/13. In macOS 14 and higher there is the new [Observation framework](https://developer.apple.com/documentation/observation) which is probably more intuitive.

Essentially, the key thing to remember is that @Published properties cannot also be computed properties, so we need to attach update handlers to them using Combine, where we update state in the callback run on the main thread.